### PR TITLE
Feat/process data

### PR DESCRIPTION
**Pull request by over-geek**
## Blog app: Processing Data

In this milestone, the requirements were:

- Create models

- Create model classes for all entities as shown in the ERD diagram. Set up associations between models.

- Use models to insert data

- Familiarize with 'rails c' command.

- Create at least one user by running the following code: first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.') second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')

- Create at least 4 posts written by one of the users you created by running the following code: first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post')

- Create at least 6 posts comments for one of the posts you created by running the following code: Comment.create(post: first_post, author: second_user, text: 'Hi Tom!' )

- Create custom methods
The models should include some additional methods.

Users

- A method that returns the 3 most recent posts for a given user. Posts
- A method that updates the posts counter for a user. A method which returns the 5 most recent comments for a given post. Comments
- A method that updates the comments counter for a post. Likes
- A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,2 +1,7 @@
 class User < ApplicationRecord
+  has_many :posts, foreign_key: 'author_id', class_name: 'Post'
```
Comment (nahnahdev) [inline app/models/user.rb:2]: - [ ] Great job so far @over-geek There are a few changes I will suggest you implement and you'll be on your way to the next Milestone :+1: 
- [ ] Per the project requirements, you are to ensure that the associations between models are set up correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png) and from the ERD diagram, you can see that the `User Table` has a one to many relationships with `Posts`, `Comments` and `Likes` but currently in the `User Table`, only the association between users and posts have been set. So kindly fix that :wink:

File: app/models/post.rb [hunk 2]
```diff
@@ -1,3 +1,21 @@
 class Post < ApplicationRecord
   belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def update_author_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
```
Comment (nahnahdev) [inline app/models/post.rb:8]: - [ OPTIONAL ] I will suggest that you consider using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) : `after_save` to call the `#update_author_posts_counter` method every time a new post has been saved to database, then make the `update_author_posts_counter` method private. 

- Example Implementation
```ruby
  # app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end

```

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @over-geek , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights :fire: 
- Has `User`, `Post`, `Comment`, and `Like` models inside `app/models/`  directory :heavy_check_mark: 
- Most of the associations between models are set up correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).The associations between models are set up correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png). :heavy_check_mark: 
- User model has a method that returns 3 most recent posts :heavy_check_mark: 
- Post model has a method that updates the posts counter for a user :heavy_check_mark: 
- Post model has a method that returns the 5 most recent comments for a given post :heavy_check_mark: 
- Comment model has a method that updates the comments counter for a post :heavy_check_mark: 
- Like model has a method that updates the likes counter for a post :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (Whoistolu) [reply app/models/user.rb:2]: - Well resolved

**Review by Whoistolu — APPROVED** [review]
### Hi @over-geek 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**